addon/decorators/track-on-constructor.ts

```diff
@@ -0,0 +1,29 @@
+import { ActivityType } from '../services/activity-tracking';
+import { assert } from '@ember/debug';
+import { getOwner } from '@ember/application';
+
+interface Constructable {
+  new (...args: any[]): any;
+}
+
+export function trackOnConstructor(actionDescription: string, actionType: ActivityType) {
```

Comment:
are we sure we want to track constructor? How does that work for  the multiple opening of the same modal, would that track it one time or each time?

Reply:
This is just yet another option that we have, we don't have any obligation to use it :D 

What do you mean by "the same modal" ? The decorator will indeed trigger on each new Ctor called, that's where the `actionDescription` & `actionType` come in handy. They will allow us to distinguish between let's say a "Confirmation modal" and a "Product selection" modal.